docs/embedding.md

```diff
@@ -2,4 +2,23 @@
 
 You can embed an EdgeSpec service inside other applications.
 
-TODO
+For example, say you want to embed an API for ships in Star Wars inside an API for all things Star Wars:
+
+```ts
+// api/ships/[...modulePath].ts
+import { withEdgeSpec } from "../with-edge-spec"
+import { z } from "zod"
+
+export default withEdgeSpec({
+  auth: "none",
+  methods: ["GET"],
+})((req) => {
+  // This is a thin wrapper over `await import()` to enable types, you can use `import * as shipsApi from "@acme/star-wars-ships` directly if you want.
+  const shipsApi = await loadBundle("@acme/star-wars-ships")
+
+  return shipsApi.makeRequest(req, {
+    // This is important so that the ships API sees the correct path.
+    removePathnamePrefix: "/ships",
```

Comment:
open to other names for this

Reply:
I thought when I did the module service implementation that this is corrected automatically

i.e. that you only see the route path relative to the module service, instead of from root

Reply:
I think it would be preferable to have it resolved automatically if possible instead of making people provide the path manually

Reply:
Hmm, interesting. See the implementation of module services and let me know if you see anything fuzzy there.

I think we can provide something like this (`removePathnamePrefix`) later for sure though for custom use cases

Reply:
ok edge case: your module service has a `[...path]/*` route at the root
I think we can still attempt to detect the base path by default though and allow overriding with a flag

Reply:
Why is that an edge case? Can't the `[...path]` merge with the `[...path]` of the parent?

Like certainly you could have

- Parent: `/path/to/module/[...path]`
- Child route `[...path]`

Then the child route just matches the parent route, right?

Reply:
no, that's not the way i implemented it for module services

for module services it substitutes so that the spread route parameter becomes the child's route

So in that case it would still 404 because the parent is `/foo/[...path]`, so the route becomes `/neste/health`, which doesn't exist in the module service causing a 404

Reply:
that assumes the parent application is using edgespec/has `.routeParams` available then? 

tbh I'm having a bit of a hard time understanding your implementation

sorry I can sense the frustration lol

Reply:
e.x. a simple greedy matcher that works in most cases:

<img width="998" alt="Screenshot 2024-02-14 at 11 48 06" src="https://github.com/seamapi/edgespec/assets/7410405/ba6718e3-6d68-46d1-b5a9-302dfde52b4a">

Reply:
^ example implementation in 3ec5cb4